### DISCUSSION TITLE
Server start priority

### DISCUSSION THREAD

**Discussion by maxcom1** [pelican-dev/panel#582] [Ideas]
So, in panel we often have game servers _and_ databases that they are connected to. If the server system (linux) is restarting, the panel tries to boot all servers that were previousley running.

Now, all servers start at the same time, so sometimes the database doesn't have enough time to finish loading before one of the game servers needs it.

We could have server start priorities, so e.g. databases start first, and then, when they started - other servers with lower priorities are booted.

**Comment by zRxnx** [comment]
Like depedencies would be cool.

Minecraft Network:
Proxy
Lobby
Survival 1
Survival 2

Survival 1/2 requires PROXY and LOBBY to be started
LOBBY needs PROXY to be started.

So if i want to start SURVIVAL1 it first needs to check if LOBBY is started if not SURVIVAL1 waits until LOBBY ist started
LOBBY needs PROXY to be started and so on

**Comment by notAreYouScared** [comment]
While this would be nice, I don't think this would be somthing we want to add.

**Comment by maxcom1** [comment]
No reason given?

**Reply by notAreYouScared** [reply]
The reason is that we don't have a interest in building our this functionality. We currently dont have functionality to auto start servers on wings boot let alone a boot order.
